src/BenchmarkDotNet/Configs/ManualConfig.cs

```diff
@@ -29,7 +29,8 @@ public class ManualConfig : IConfig
         private readonly List<Job> jobs = new List<Job>();
         private readonly HashSet<HardwareCounter> hardwareCounters = new HashSet<HardwareCounter>();
         private readonly List<IFilter> filters = new List<IFilter>();
-        private readonly List<BenchmarkLogicalGroupRule> logicalGroupRules = new List<BenchmarkLogicalGroupRule>();
+        private readonly HashSet<BenchmarkLogicalGroupRule> logicalGroupRules = new HashSet<BenchmarkLogicalGroupRule>();
+        private readonly static Conclusion[] emptyConclusion = Array.Empty<Conclusion>();
```

Comment:
@AndreyAkinshin @workgroupengineering
Not sure if you noticed, but this turns logicalGroupRules back into HashSet<> (changed to List<> in #1866).
(I didn't do a full review, but this one jumped at me)

Reply:
@mawosoft nice catch! Fixed in bd08722ee22fc25a0cf6d389125c5b6eff10cd8f